src/librbd/watcher/Types.cc

```diff
@@ -6,6 +6,9 @@
 #include "common/Formatter.h"
 #include "common/dout.h"
 
+#include "librbd/ImageCtx.h"
+#include "librbd/MirroringWatcher.h"
```

Comment:
Since this is the generic types class -- it shouldn't include the derived class.